tests/queries/0_stateless/03732_join_empty_optimization.sql.j2

```diff
@@ -0,0 +1,142 @@
+{% set join_kinds = ['INNER', 'LEFT', 'RIGHT', 'FULL', 'CROSS'] %}
```

Comment:
Overall, the test is alright, but a bit overcomplicated. 

We could just do a select with `count` or `countIf` from `EXPLAIN`, expecting the query to return `1` (or the number you need, depending on the check). 

If you want to make the reference more verbose, you may also set `-- { echoOn }` before select queries, e.g., https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/0_stateless/03357_join_pk_sharding.reference#L15-L29

Also, `scenario` and other variables make it difficult to read the queries. I'd just make a loop on variables that are easy to set in the query (like storages, join kinds), but write queries explicitly without the macro.